src/common/AdaptiveTree.cpp

```diff
@@ -86,9 +86,6 @@ namespace adaptive
       if (rep->flags_ & Representation::URLSEGMENTS)
         delete[] bs->url;
     }
-    if ((rep->flags_ & (Representation::INITIALIZATION | Representation::URLSEGMENTS))
-      == (Representation::INITIALIZATION | Representation::URLSEGMENTS))
-      delete[]rep->initialization_.url;
```

Comment:
Why is this removed?

Reply:
With only byteranged initialization support beforehand (no URLSEGMENTS) this code block would have never been executed (as far as I can see it is only called from HLSTree.cpp). Now with fmp4 urlsegments it deletes the initialization url from the representation causing the stream to fail to play.
Would you prefer instead to override the function in HLSTree?

Reply:
Also once the initialization url is set there should be no need to change or delete it, it should survive for the life of the representation. Further EXT-X-MAP tags will also be accompanied by EXT-X-DISCONTINUITY tags which mean new period/new representation.

Reply:
First this is the general implementation for all parser, and therefore should never modified because it is currently called only from HLS tree, but from what I read in your code (see new block #EXT-X-MAP)  the HLS initialization segment uses _url and allocates data, so it must be deleted[] to avoid memory leaks.
What is the real issue if the code abouve is not deleted?

Reply:
Ok I understand I shouldn't modify the general implementation to suit only one of dash/hls/ss, once we figure a way forward I will adjust accordingly.

From what I understand FreeSegments is there to clean up any allocated data on playlist refreshes before newSegments is swapped into rep->segments_. The real issue with the code in FreeSegments is that rep->initialization->url_ is being deleted pretty much straight after being set and then shortly after in [restart_stream()](https://github.com/peak3d/inputstream.adaptive/blob/master/src/common/AdaptiveStream.cpp#L224) the init download fails e.g in windows the log shows `2020-01-17 23:31:56.284 T:17816   ERROR: CWIN32Util::ConvertPathToWin32Form: Error converting path "ޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝޝa" to Win32 form` - we're reading garbage from memory for the url_.

Another way to put it - there's no 'newInitialization' that's being swapped into the rep after FreeSegments is called like in the case of newSegments. Is that a solution that you'd prefer to see btw?

It's not an issue for byteranged fmp4 as there is no rep->initialization->url_ needed or set.

Reply:
Thx for clarification @glennguy . I would prefer the newInitialization in this case.
If you assign it after the FreeSegments call you dont have to care about freeing the existing memory (in case you revert your before mentioned changes in AdaptiveTree)

Reply:
Thanks for your guidance, I think it's good now. Could you review the latest changes @peak3d?